### DISCUSSION TITLE
Buttons not registering based on AsyncEventButton example

### DISCUSSION THREAD

**Discussion by shampeon** [vortigont/ESPAsyncButton#11] [Q&A]
I'm trying to get my sketch, which was working before using the raw pin interface, to work with long presses, and found this library. It's using an ESP32, and some momentary buttons on pins 17 and 19.

I'm following the `AsyncEventButton` example:

``` c++
// buttons
AsyncEventButton zeroButton(GPIO_NUM_17, LOW);
AsyncEventButton unitButton(GPIO_NUM_19, LOW);
```

In `setup()`:

``` c++
void setup()
{
  ESP_ERROR_CHECK(esp_event_loop_create_default());
...
  zeroButton.begin();
  zeroButton.onPress(onZeroPressed);
  zeroButton.onLongPress(onZeroLongPressed);
  unitButton.begin();
  unitButton.onPress(onUnitPressed);
  unitButton.onLongPress(onUnitLongPressed);
  unitButton.onMultiClick(onUnitMulticlick);
...
}
```

One of the callbacks:

``` c++
// button callbacks
void onZeroPressed() {
  Serial.println("Zero button pressed");
  tareValue = convertedValue; // use the most recent conversion value as the tare value
  display.ssd1306_command(SSD1306_DISPLAYON);
  sleepTimerTask.restartDelayed();
}
```

I don't get any messages on pressing either button in the serial console. I've also tried changing the event to `onClick` for the button. Nothing registers. I've simplified the sketch to only define one button, but that also doesn't work.

Again, this sketch was working before using the same pins and reading the pin state with `digitalRead`. So the wiring is correct. I do not see what I'm doing wrong here.

**Comment by shampeon** [comment]
Figures I would find it right after asking. I didn't call `.enable()` within `setup()` on the buttons.....

**Reply by vortigont** [reply]
Nice to hear you figured it out. Yep, `.enable()` is also needed to activate interrupts on a gpio.
